src/extension.ts

```diff
@@ -68,7 +69,9 @@ export async function activate(context: vscode.ExtensionContext) {
  * Deactivate the extension.
  * 
  * Any disposables registered in `context.subscriptions` will be automatically
- * disposed of, so there's nothing left to do here.
+ * disposed of.
  */
-export function deactivate() {}
+export function deactivate() {
+	statusItem.dispose();
```

Comment:
Any reason why you aren't just adding this to context.subscriptions. You'd need to conform `StatusItem` to `Disposable`.

Reply:
No particular reason. I wasn't sure where to place the global `statusItem` object, and since it isn't created in the activation function, I just disposed of it here.

Reply:
You can't have it as a global. If you load a new workspace the item will have dispose called on it when deactivate is called when shutting down the first workspace. Then for the new workspace you will have a disposed StatusItem. 

You need to create a new StatusItem every time activate is called. Maybe you can store the StatusItem in the context, so it can be accessed. Also maybe wait for #32 as that changes the structure of the context.

Reply:
Hmm, when I add logging to the constructor of the `statusItem`, I can see that VS Code recreates the extension (and the `statusItem`):
- When I go from a single folder to an unnamed workspace
- When I remove the first (top most) folder from a workspace

This is also mentioned in the API docs for `onDidChangeWorkspaceFolders`:

> onDidChangeWorkspaceFolders: Event<WorkspaceFoldersChangeEvent>
> 
> An event that is emitted when a workspace folder is added or removed.
> 
> Note: this event will not fire if the first workspace folder is added, removed or changed, because in that case the currently executing extensions (including the one that listens to this event) will be terminated and restarted so that the (deprecated) rootPath property is updated to point to the first workspace folder.

Reply:
Yeah moving between single folder workspace to multi-folder workspace creates a new session. I got slightly thrown by that when working on #32. 

Still the StatusItem lifecycle should be linked to the lifecycle of the session.